packages/zwave-js/src/lib/commandclass/HumidityControlSetpointCC.ts

```diff
@@ -0,0 +1,939 @@
+	// We expect a Thermostat Setpoint Report that matches the requested setpoint type
+	return received.type === sent.setpointType;
+}
+
+interface HumidityControlSetpointCCGetOptions extends CCCommandOptions {
+	setpointType: HumidityControlSetpointType;
+}
+
+@CCCommand(HumidityControlSetpointCommand.Get)
+@expectedCCResponse(
+	HumidityControlSetpointCCReport,
+	testResponseForHumidityControlSetpointGet,
+)
+export class HumidityControlSetpointCCGet extends HumidityControlSetpointCC {
+	public constructor(
+		driver: Driver,
+		options:
+			| CommandClassDeserializationOptions
+			| HumidityControlSetpointCCGetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			// TODO: Deserialize payload
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			this.setpointType = options.setpointType;
+		}
+	}
+
+	public setpointType: HumidityControlSetpointType;
+
+	public serialize(): Buffer {
+		this.payload = Buffer.from([this.setpointType & 0b1111]);
+		return super.serialize();
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		return {
+			...super.toLogEntry(),
+			message: {
+				"setpoint type": getEnumMemberName(
+					HumidityControlSetpointType,
+					this.setpointType,
+				),
+			},
+		};
+	}
+}
+
+@CCCommand(HumidityControlSetpointCommand.SupportedReport)
+export class HumidityControlSetpointCCSupportedReport extends HumidityControlSetpointCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 1);
+		this._supportedSetpointTypes = parseBitMask(
+			this.payload,
+			HumidityControlSetpointType["N/A"],
+		);
+
+		this.persistValues();
+	}
+
+	private _supportedSetpointTypes: HumidityControlSetpointType[];
+	@ccValue({ internal: true })
+	public get supportedSetpointTypes(): readonly HumidityControlSetpointType[] {
+		return this._supportedSetpointTypes;
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		return {
+			...super.toLogEntry(),
+			message: {
+				"supported setpoint types": this.supportedSetpointTypes
+					.map(
+						(t) =>
+							`\n· ${getEnumMemberName(
+								HumidityControlSetpointType,
+								t,
+							)}`,
+					)
+					.join(""),
+			},
+		};
+	}
+}
+
+@CCCommand(HumidityControlSetpointCommand.SupportedGet)
+@expectedCCResponse(HumidityControlSetpointCCSupportedReport)
+export class HumidityControlSetpointCCSupportedGet extends HumidityControlSetpointCC {}
+
+@CCCommand(HumidityControlSetpointCommand.ScaleSupportedReport)
+export class HumidityControlSetpointCCScaleSupportedReport extends HumidityControlSetpointCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 1);
+
+		const supportedScaleIndices = parseBitMask(this.payload, 0);
```

Comment:
This should make sure not to parse any unrelated bits, e.g. like this:
```suggestion
		const supportedScaleIndices = parseBitMask(Buffer.from([this.payload[0] & 0b1111]), 0);
```